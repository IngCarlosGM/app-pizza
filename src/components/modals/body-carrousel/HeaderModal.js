import React from 'react';
import { Link } from 'react-router-dom';
import queryString from 'query-string';

const HeaderModal = ({ currentItem= 1, storeToShow, history }) => {

    const params = queryString.parse(history.location.search);
    let q = (params.q) ? `?q=${params.q}`: '';        

    return (
        <div className="container__modal-header space-between">
            <h5 className="container__modal-header-title">{storeToShow.name}</h5>
            <img 
                className="modal-header__logo-store"
                src={`../assets/store/store-${ currentItem }.png`} 
                alt="Logo pizza" 
            />
            <Link 
                to={`/${ q }`}
                className="modal-header__close-modal pointer"
            >x</Link>
        </div>
    )
}

export default HeaderModal;
