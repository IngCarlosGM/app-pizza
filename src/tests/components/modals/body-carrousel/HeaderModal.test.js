import { shallow } from "enzyme";
import HeaderModal from "../../../../components/modals/body-carrousel/HeaderModal";
import dataApp from '../../../data/dataApp.json';

describe('Testing component HeaderModal', () => {

    const history = {
        location: {
            search: "url-test"
        }
    }
    
    const wrapper = shallow(
        <HeaderModal
            currentItem= { 1 }
            storeToShow= { dataApp.response.stores[0] }
            history= { history }
        />
    );

    test('should render correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should to have a modal title', () => {
        expect( wrapper.exists('h5') ).toBe( true );
        expect( wrapper.find('h5').at(0).text() ).toBe( dataApp.response.stores[0].name );
    });

    test('should to have a image', () => {
        expect( wrapper.exists('img') ).toBe( true );
    });

    test('should to have a image correct', () => {
        expect( wrapper.find('img').at(0).prop('src') ).toBe( `../assets/store/store-${ dataApp.response.stores[0].id }.png` );
    });
    
});
