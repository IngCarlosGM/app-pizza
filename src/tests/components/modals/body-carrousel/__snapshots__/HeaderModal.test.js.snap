// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component HeaderModal should render correctly 1`] = `
<div
  className="container__modal-header space-between"
>
  <h5
    className="container__modal-header-title"
  >
    Pizzeria Capitan America
  </h5>
  <img
    alt="Logo pizza"
    className="modal-header__logo-store"
    src="../assets/store/store-1.png"
  />
  <Link
    className="modal-header__close-modal pointer"
    to="/"
  >
    x
  </Link>
</div>
`;
